File: reportcompiler_debugging_tools/python/debug.py
import json
import os
import sys
import pandas as pd
from reportcompiler.plugins.source_parsers.python \
    import PythonParser
from reportcompiler.plugins.source_parsers.base \
    import SourceParser


def generate_failed_contexts(report_path):
    try:
        with open(os.path.join(report_path,
                               '_meta',
                               'last_debug_errors'), 'r') as f:
            last_values = json.load(f)
    except FileNotFoundError:
        raise FileNotFoundError(
            'There is no file with debugging data, please retry '
            'generating a document on debug mode')

    for frag in last_values:
        print("Error on fragment '{}' ({})".format(
            frag['metadata']['fragment_name'],
            frag['timestamp']))
        doc_var = frag['doc_var']
        data = {name: pd.DataFrame(data)
                for name, data in frag['data'].items()}
        metadata = frag['metadata']

        fragment = os.path.basename(metadata['fragment_path'])
        _, ext = os.path.splitext(fragment)

        generator_info = metadata.get('context_generator')
        if isinstance(generator_info, str):
            generator = SourceParser.get(id=generator_info,
                                         extension=ext)
        elif isinstance(generator_info, dict):
            generator = SourceParser.get(
                            id=generator_info[ext])

        if generator.__class__ == PythonParser:
            sys.path.append(os.path.join(report_path, frag['report'], 'src'))
            try:
                ctx = PythonParser().generate_context(doc_var,
                                                      data,
                                                      metadata)
                print('\n{}\n'.format(ctx))
                print('*** {} has finished successfully.'.format(fragment))
            except Exception:
                print('*** {} has finished with errors.'.format(fragment))
        else:
            print("*** {} doesn't use python, skipping...".format(fragment))


if __name__ == '__main__':
    generate_failed_contexts(report_path=os.environ['REPORTS_PATH'])

File: reportcompiler_debugging_tools/r/debug.r
library(jsonlite)

compile_last_fragment <- function(report_path) {
    tryCatch(
        json_file <- file(paste(report_path, '_meta', 'last_debug_errors',sep='/'), open='r'),
        error=function() {
            print('There is no file with debugging data, please retry generating a document on debug mode')
        })
    last_values <- fromJSON(json_file, simplifyVector=F)
    
    for(frag in last_values) {
        doc_var <- frag$doc_var
        data <- lapply(names(frag$data), function(n){data.frame(frag$data[n][[1]])})
        names(data) <- names(frag$data)
        metadata <- frag$metadata
        
        path <- metadata$fragment_path
        if (any(endsWith(path, c('.R', '.r')))) {
            source(metadata$fragment_path)
            tryCatch({
                  generate_context(doc_var, data, metadata)
                },
                error=function(){
                    print(paste(basename(path), " has finished with errors."))
                }
            )
            print(paste(basename(path), " has finished successfully."))
        } else {
            print(paste(basename(path), " doesn't use R, skipping..."))
        }
    }
}

reports.path <- Sys.getenv('REPORTS_PATH')

compile_last_fragment(reports.path)